test/components/linkerd/linkerd_test.go

```diff
@@ -0,0 +1,233 @@
+  - image: fedora
+    name: download-linkerd
+    command:
+    - bash
+    args:
+    - -c
+    - 'curl -L https://run.linkerd.io/install | sh && cp -L /root/.linkerd2/bin/linkerd /data/linkerd'
+    volumeMounts:
+    - name: download-dir
+      mountPath: /data
+  containers:
+  - image: fedora
+    name: linkerd-cli
+    env:
+    - name: KUBECONFIG
+      value: /root/.kube/config
+    command:
+    - bash
+    args:
+    - -c
+    - 'mv /data/* /usr/local/bin/ && kubectl version -o yaml && linkerd version && linkerd check'
+    volumeMounts:
+    - name: kubeconfig
+      mountPath: /root/.kube
+    - name: download-dir
+      mountPath: /data
+  volumes:
+  - name: kubeconfig
+    secret:
+      secretName: kubeconfig
+  - name: download-dir
+    emptyDir: {}
+`
+
+	namespaceConfig = `
+apiVersion: v1
+kind: Namespace
+metadata:
+  generateName: linkerd-check-test-
+`
+
+	retryInterval = 5 * time.Second
+	timeout       = 9 * time.Minute
+)
+
+// nolint: funlen
+func TestLinkerdCheck(t *testing.T) {
+	t.Parallel()
+
+	client := testutil.CreateKubeClient(t).CoreV1()
+	nsclient := client.Namespaces()
+
+	// Parsing
+	ns := &corev1.Namespace{}
+	if err := yaml.Unmarshal([]byte(namespaceConfig), ns); err != nil {
+		t.Fatalf("failed to unmarshal namespace manifest: %v", err)
+	}
+
+	pod := &corev1.Pod{}
+	if err := yaml.Unmarshal([]byte(podConfig), pod); err != nil {
+		t.Fatalf("failed to unmarshal pod manifest: %v", err)
+	}
+
+	// Creation
+	ns, err := nsclient.Create(context.TODO(), ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("failed creating Namespace: %v", err)
+	}
+
+	t.Cleanup(func() {
+		if err := nsclient.Delete(context.TODO(), ns.Name, metav1.DeleteOptions{}); err != nil {
+			t.Logf("failed removing Namespace: %v", err)
+		}
+	})
+
+	_, err = client.Secrets(ns.Name).Create(context.TODO(), getKubeconfigSecret(t), metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("failed creating ConfigMap: %v", err)
+	}
+
+	podsClient := client.Pods(ns.Name)
+
+	// Retry pod creation. This might fail if Linkerd is not ready yet and some requests might fail.
+	if err := wait.PollImmediate(retryInterval, timeout, func() (done bool, err error) {
+		pod, err = podsClient.Create(context.TODO(), pod, metav1.CreateOptions{})
+		if err != nil {
+			t.Logf("retrying pod creation, failed with: %v", err)
+			return false, nil
+		}
+
+		return true, nil
+	}); err != nil {
+		t.Fatalf("error while trying to create the pod: %v", err)
+	}
+
+	phase := corev1.PodUnknown
+
+	if err := wait.PollImmediate(retryInterval, timeout, func() (done bool, err error) {
+		p, err := podsClient.Get(context.TODO(), pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		phase = p.Status.Phase
+		if phase == corev1.PodSucceeded || phase == corev1.PodFailed {
+			return true, nil
+		}
+
+		return false, nil
+	}); err != nil {
+		t.Errorf("error while waiting for the pod: %v", err)
+	}
+
+	// Since pod failed print the logs
```

Comment:
Also, why not do that in the block above?

```suggestion
	// Since pod failed print the logs.
```